tests/lean/interactive/hover.lean

```diff
@@ -244,3 +244,7 @@ example : Nat → Nat
     --v textDocument/hover
   | .succ x => .succ x
                --^ textDocument/hover
+
+example : Inhabited Nat := ⟨Nat.zero⟩
+                         --^ textDocument/hover
+                          --^ textDocument/hover
```

Comment:
Can you add an example for hovering `(. + .)`? I think that was the example that was reported.

Reply:
That's the separate issue about original ranges becoming (non-canonical) synthetic, whose fix I removed again

Reply:
Actually would it help if `expandCDot?` turned the original ranges into canonical ones? That *sounds* sensible.

Reply:
Ok, looks pretty good! We lost the elaborator docstring on `·`, but that seems fair in exchange for the added type information. Not quite sure how to achieve both at the same time, except by applying the docstring to a copy of `elabIdent`...